clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1362,7 +1402,7 @@ class SyclKernelBodyCreator
   bool handleSpecialType(FieldDecl *FD, QualType Ty) {
     const auto *RecordDecl = Ty->getAsCXXRecordDecl();
     // Perform initialization only if it is field of kernel object
-    if (MemberExprBases.size() == 1) {
```

Comment:
I think this just matches with previous behavior, we didn't initialize anything except kernel object fields, we don't do this now. If we start arrays initialization by removing this line, this will cause crashes, since this function is intended to initialize fields of structs and not array elements, and we enter it trying to initialize array elements.